src/modules/fancyzones/editor/FancyZonesEditor/Models/CanvasLayoutModel.cs

```diff
@@ -91,6 +91,25 @@ public void AddZone()
             UpdateLayout();
         }
 
+        public void ScaleLayout(double workAreaWidth, double workAreaHeight)
+        {
+            Int32Rect[] zones = new Int32Rect[Zones.Count];
+            Zones.CopyTo(zones, 0);
+            Zones.Clear();
+
+            foreach (Int32Rect zone in zones)
+            {
+                var x = zone.X * workAreaWidth / CanvasRect.Width;
+                var y = zone.Y * workAreaHeight / CanvasRect.Height;
+                var width = zone.Width * workAreaWidth / CanvasRect.Width;
+                var height = zone.Height * workAreaHeight / CanvasRect.Height;
```

Comment:
Isn't there some possible division by 0s here?

Reply:
You're right. It can happen only if `ref-width` or `ref-height` in the json file is 0, but overall it can happen, so the check is needed.